sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -360,6 +360,15 @@ def visit_transform(self, transform_node):
 
   def run_pipeline(self, pipeline, options):
     """Remotely executes entire pipeline or parts reachable from node."""
+    # Label goog-notebook if pipeline is initiated from interactive runner.
+    from apache_beam.runners.interactive import interactive_runner
+    if isinstance(pipeline.runner, interactive_runner.InteractiveRunner):
```

Comment:
This seems fine - but what if we go with the `runner_in_use` codepath? Would users do: `p.run(runner=InteractiveRunner(DataflowRunner()), options=...)`? Or would users create a pipeline with InteractiveRunner and then do `p.run(runner=DataflowRunner()...`? Is it poissible for users to do `p = beam.Pipeline()`, and then do `InteractiveRunner().run_pipeline(p)`/`InteractiveRunner(DataflowRunner()).run_pipeline(p)`?

IIUC users would have to pass the interactive runner in `p = beam.Pipeline()` to activate the interactive mode, right? InteractiveRunner is not automatically selected?

Reply:
I've missed the path where a new Pipeline is created and `run()` is invoked again.
Yes, all of these would be possible.
I've added an `interactive` parameter at the constructor level for `Pipeline` using default value `None`. `run()` and `from_runner_api()` will pass the `None` or `bool` value down no matter how the user chains the runners. I'm not very confident with the naming but the change should be backward compatible for Beam.

Currently, I'm running into a problem when testing. Once I set `labels`, Dataflow job will fail immediately and throw `Error processing pipeline.` error. There will be no job graph, no worker started, no logs. Looks like when there is user label in the job request, Dataflow cannot convert the work item into internal representation.

I'll do some investigation and figure out why.